### PR TITLE
Add raw address to AsyncPlayerPreLoginEvent

### PR DESCRIPTION
**Pull request by ConnorLinfoot**
Would be nice to be able to access the raw address during this event instead of having to wait and check on the PlayerLoginEvent, especially if you wish to do something based on that information in AsyncPlayerPreLoginEvent.

### CODE REVIEW

File: Spigot-Server-Patches/0726-Add-raw-address-to-AsyncPlayerPreLoginEvent.patch [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+diff --git a/src/main/java/net/minecraft/server/network/LoginListener.java b/src/main/java/net/minecraft/server/network/LoginListener.java
+index c67b94840e4c967baebf6eb351df15f0e4ead4be..933bc5231d7990a6732224092f9bfb9d5f5e7112 100644
+--- a/src/main/java/net/minecraft/server/network/LoginListener.java
++++ b/src/main/java/net/minecraft/server/network/LoginListener.java
+@@ -327,12 +327,13 @@ public class LoginListener implements PacketLoginInListener {
+                         // Paper end
+                         String playerName = i.getName();
+                         java.net.InetAddress address = ((java.net.InetSocketAddress) networkManager.getSocketAddress()).getAddress();
++                        java.net.InetAddress rawAddress = ((java.net.InetSocketAddress) networkManager.getRawAddress()).getAddress();
```
Comment (MiniDigger) [inline Spigot-Server-Patches/0726-Add-raw-address-to-AsyncPlayerPreLoginEvent.patch]: paper comment

File: Spigot-API-Patches/0297-Add-raw-address-to-AsyncPlayerPreLoginEvent.patch [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+     private final InetAddress ipAddress;
++    private final InetAddress rawAddress; // Paper
+     //private UUID uniqueId; // Paper - Not used anymore
+ 
+     @Deprecated
+@@ -49,7 +50,20 @@ public class AsyncPlayerPreLoginEvent extends Event {
+         this.profile = profile;
+     }
+ 
++    /**
```
Comment (MiniDigger) [inline Spigot-API-Patches/0297-Add-raw-address-to-AsyncPlayerPreLoginEvent.patch:24]: paper comments

**Review by MiniDigger — APPROVED** [review]
editing a patch file by hand is naughty and will break it if you add new lines :P

we can handle that and the missing paper comments while merging tho, thanks for the contribution!

**Comment by ConnorLinfoot** [comment]
Resolved the patch issue and added the missing comments :)

**Review by kennytv — APPROVED** [review]
Thank you

**Pull request merged**